src/tools/unpack.c

```diff
@@ -937,32 +939,38 @@ void unpack_cleanup(void *user_data)
     for (int i=0; i<opts->n_sourcedirs; i++) {
         FREE_VAR(opts->sourcedirs[i]);
     }
+
+    sfmf_control_close();
 }
 
 static void draw_progress(struct UnpackOptions *opts, int i, const char *message)
 {
     static float last_progress = -1.f;
 
-    if (opts->progress) {
-        float partial = (opts->header.entries_length && i >= 0)
-            ? fminf(1.f, (float)(i) / (float)(opts->header.entries_length))
-            : (i == -1 ? 0.f : 1.f);
+    float partial = (opts->header.entries_length && i >= 0)
+        ? fminf(1.f, (float)(i) / (float)(opts->header.entries_length))
+        : (i == -1 ? 0.f : 1.f);
```

Comment:
It's actually called with "-2" to mean "finished", so I'll add the following comment on top (it is not at all obvious that this is the case without context).

```
// i == -1 means "starting", i == -2 means "finished"
```